compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/MockDesugar.java

```diff
@@ -160,13 +253,14 @@ private BInvokableSymbol generateSymbol(String functionName) {
     }
 
     private BInvokableType generateSymbolInvokableType() {
-        BInvokableType bInvokableType = (BInvokableType) this.originalFunction.symbol.type;
-        return bInvokableType;
-    }
+        BInvokableType bInvokableType;
 
-    private BType generateType() {
-        BType type = ((BInvokableType) this.originalFunction.type).retType;
-        return type;
+        if (this.originalFunction == null) {
+            bInvokableType = (BInvokableType) this.importFunction.type;
```

Comment:
Is "this.importFunction" nullable? Should we add a null check for that as well?

Reply:
I added a null check for both import function and original function. In this case, one of the two has to be null. If both arent then the function will not go through the desugaring process and throws an error instead.